geomag_imp/secs.py

```diff
@@ -518,146 +522,141 @@ def make_T(self, lat_lon_r):
+         Btheta2 = np.divide(Btheta2, sin_theta, out=np.zeros_like(sin_theta),
+                              where=sin_theta != 0)
+
+         # Update only the locations where secs are under observations
+         Btheta[under_locs] = Btheta2[under_locs]
+         Br[under_locs] = Br2[under_locs]
+
+      # Transform back to Bx, By, Bz at each local point
+      T = np.empty((nobs, 3, nsec))
+      # alpha == angle (from cartesian x-axis (By), going towards y-axis (Bx))
+      T[:, 0, :] = Btheta*np.sin(alpha)
+      T[:, 1, :] = Btheta*np.cos(alpha)
+      T[:, 2, :] = Br
+            
+      return T.reshape(nobs * 3, -1)
+
+
+def calc_angular_distance(latlon1, latlon2):
+   """Calculate the angular distance between a set of points.
+   This function calculates the angular distance in radians
+   between any number of latitude and longitude points.
+   Parameters
+   ----------
+   latlon1 : ndarray (n x 2 [lat, lon])
+      An array of n (latitude, longitude) points.
+   latlon2 : ndarray (m x 2 [lat, lon])
+      An array of m (latitude, longitude) points.
+   Returns
+   -------
+   ndarray (n x m)
+      The array of distances between the input arrays.
+   """
+   lat1 = np.deg2rad(latlon1[:, 0])[:, np.newaxis]
+   lon1 = np.deg2rad(latlon1[:, 1])[:, np.newaxis]
+   lat2 = np.deg2rad(latlon2[:, 0])[np.newaxis, :]
+   lon2 = np.deg2rad(latlon2[:, 1])[np.newaxis, :]
+
+   dlon = lon2 - lon1
+
+   # theta == angular distance between two points
+   theta = np.arccos(np.sin(lat1)*np.sin(lat2) +
+                     np.cos(lat1)*np.cos(lat2)*np.cos(dlon))
+   return theta
+
+
+def calc_bearing(latlon1, latlon2):
+   """Calculate the bearing (direction) between a set of points.
+   This function calculates the bearing in radians
+   between any number of latitude and longitude points.
+   It is the direction from point 1 to point 2 going from the
+   cartesian x-axis towards the cartesian y-axis.
+   Parameters
+   ----------
+   latlon1 : ndarray (n x 2 [lat, lon])
+      An array of n (latitude, longitude) points.
+   latlon2 : ndarray (m x 2 [lat, lon])
+      An array of m (latitude, longitude) points.
+   Returns
+   -------
+   ndarray (n x m)
+      The array of bearings between the input arrays.
+   """
+   lat1 = np.deg2rad(latlon1[:, 0])[:, np.newaxis]
+   lon1 = np.deg2rad(latlon1[:, 1])[:, np.newaxis]
+   lat2 = np.deg2rad(latlon2[:, 0])[np.newaxis, :]
+   lon2 = np.deg2rad(latlon2[:, 1])[np.newaxis, :]
+
+   dlon = lon2 - lon1
+
+   # alpha == bearing, going from point1 to point2
+   #          angle (from cartesian x-axis (By), going towards y-axis (Bx))
+   # Used to rotate the SEC coordinate frame into the observation coordinate
+   # frame.
+   # SEC coordinates are: theta (colatitude (+ away from North Pole)),
+   #                      phi (longitude, + east), r (+ out)
+   # Obs coordinates are: X (+ north), Y (+ east), Z (+ down)
+   alpha = np.pi/2 - np.arctan2(np.sin(dlon)*np.cos(lat2),
```

Comment:
I suppose if it is easier for your code, you could avoid this opposite direction for bearing here so that your angles align directly with your coordinate frame.